File: Problems/0100_0199/0106_Construct_Binary_Tree_from_Inorder_and_Postorder_Traversal/Project_Go/solution/mylib.go
package solution

import (
	"strconv"
	"strings"
	"unsafe"
)

func StringToIntArray(flds string) []int {
	numsStr := strings.Split(flds, ",")
	nums := make([]int, len(numsStr))

	for i := 0; i < len(nums); i++ {
		nums[i], _ = strconv.Atoi(numsStr[i])
	}

	return nums
}

func StringToByteArray(flds string) []byte {
	nums := make([]byte, len(flds))

	for i := 0; i < len(nums); i++ {
		nums[i] = (byte)(flds[i])
	}

	return nums
}

func IntArrayToString(nums []int) string {
	if len(nums) <= 0 {
		return ""
	}

	resultStr := strconv.Itoa(nums[0])
	for i := 1; i < len(nums); i++ {
		resultStr += ", " + strconv.Itoa(nums[i])
	}

	return resultStr
}

func IntIntArrayToString(nums [][]int) string {
	if len(nums) <= 0 {
		return "[]"
	}

	resultStr := "[[" + IntArrayToString(nums[0]) + "]"

	for i := 1; i < len(nums); i++ {
		resultStr += ", [" + IntArrayToString(nums[i]) + "]"
	}
	return resultStr + "]"
}

func IntIntArrayToGridString(grid [][]int) string {
	if len(grid) <= 0 {
		return "[[]]"
	}

	resultStr := "[\n"
	resultStr += "  [" + IntArrayToString(grid[0]) + "]\n"
	for i := 1; i < len(grid); i++ {
		resultStr += ", [" + IntArrayToString(grid[i]) + "]" + "\n"
	}
	return resultStr + "]"
}

func ByteByteArrayToString(grid [][]byte) string {
	if len(grid) <= 0 {
		return "[[]]"
	}

	resultStr := "[[" + *(*string)(unsafe.Pointer(&grid[0])) + "]"
	for i := 1; i < len(grid); i++ {
		resultStr += ", [" + *(*string)(unsafe.Pointer(&grid[i])) + "]"
	}
	return resultStr + "]"
}

func ByteByteArrayToGridString(grid [][]byte) string {
	if len(grid) <= 0 {
		return "[[]]"
	}

	resultStr := "[\n"
	resultStr += "  [" + *(*string)(unsafe.Pointer(&grid[0])) + "]\n"
	for i := 1; i < len(grid); i++ {
		resultStr += ", [" + *(*string)(unsafe.Pointer(&grid[i])) + "]\n"
	}
	return resultStr + "]"
}

func StringArrayToString(data []string) string {
	if len(data) <= 0 {
		return ""
	}

	resultStr := "\"" + data[0] + "\""
	for i := 1; i < len(data); i++ {
		resultStr += ", \"" + data[i] + "\""
	}

	return resultStr
}

func StringStringArrayToString(data [][]string) string {
	if len(data) <= 0 {
		return "[]"
	}

	res := ""
	for i := 0; i < len(data); i++ {
		res += "[\n"
		for j := 0; j < len(data[i]); j++ {
			if j == 0 {
				res += " [" + data[i][j] + "]\n"
			} else {
				res += ",[" + data[i][j] + "]\n"
			}
		}
		res += "]\n"
	}

	return res
}

File: Problems/Scripts_Replace_File/sourcefile/Mylib.java
import java.util.*;

public class Mylib {
    public int[] stringToIntArray(String s) {
        if (s.length() <= 0)
            return null;

        String[] flds = s.split(",");
        int[] nums = new int[flds.length];

        if (flds.length <= 0)
            return nums;

        for (int i = 0; i < nums.length; ++i) {
            nums[i] = Integer.parseInt(flds[i]);
        }

        return nums;
    }

    public String intArrayToString(int[] nums) {
        if (nums == null)
            return "[]";
        if (nums.length <= 0)
            return "[]";

        StringBuilder resultStr = new StringBuilder("[" + Integer.toString(nums[0]));
        for (int i = 1; i < nums.length; ++i)
            resultStr.append("," + Integer.toString(nums[i]));
        resultStr.append("]");

        return resultStr.toString();
    }

    private String stringArrayToString(String[] flds) {
        if (flds.length <= 0)
            return "[]";

        StringBuilder sb = new StringBuilder("[\"" + flds[0] + "\"");
        for (int i = 1; i < flds.length; i++)
            sb.append(", \"" + flds[i] + "\"");
        sb.append("]");

        return sb.toString();
    }

    public String listArrayToString(List<Integer> list) {
        if (list.size() <= 0)
            return "[]";

        StringBuilder sb = new StringBuilder("[" + Integer.toString(list.get(0)));
        for (int i = 1; i < list.size(); i++) {
            sb.append("," + Integer.toString(list.get(i)));
        }
        sb.append("]");

        return sb.toString();
    }

    public String listListArrayToString(List<List<Integer>> list) {
        if (list.size() <= 0)
            return "[]";

        StringBuilder sb = new StringBuilder("[" + listArrayToString(list.get(0)));
        for (int i = 1; i < list.size(); i++) {
            sb.append("," + listArrayToString(list.get(i)));
        }
        sb.append("]");

        return sb.toString();
    }
}
